deployment/frontend/src/components/dashboard/datasets/admin/EditDatasetForm.tsx

```diff
@@ -193,7 +193,7 @@ export default function EditDatasetForm({ dataset }: { dataset: WriDataset }) {
                 `Successfully edited the "${title ?? name}" dataset`,
                 'success'
             )
-            router.push('/dashboard/datasets')
+            window.location.href = '/dashboard/datasets';
```

Comment:
Only way I managed to make the redirect work. `router.push`simply wasn't working when a new layer was being added. It would change the URL in the browser but wouldn't render the new page.